File: util/CodingStyle.md/formatter/test/line_split_operator_priority_out.ts
/*
 * Copyright (C) 2024 Example Corp.
 * SPDX-License-Identifier: MIT
 */

//%JXM_CFMT_CFG line-split-operator-priority=on

function optionalChainAndNullishNeverMistakenForTernary(x: { y?: { z: number } } | null): number
{
    const totalResultValueHereXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX = someVeryLongOptionalChainExpressionNameAAAAAAAAAAAAAAAAAAAAA?.someVeryLongPropertyNameBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB ?? anotherVeryLongDefaultExpressionNameCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC;
    return totalResultValueHereXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX;
}

function genuineTernaryStillSplitsWhenTooLong(cond: boolean): string
{
    return someConditionVariableNameHereXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
        ? someVeryLongTrueValueExpressionHereAAAAAAAAAAAAAAAAAAAAAA
        : someVeryLongFalseValueExpressionHereBBBBBBBBBBBBBBBBBBBBB;
}

File: util/CodingStyle.md/formatter/test/line_split_operator_priority_out.kt
/*
 * Copyright (C) 2024 Example Corp.
 * SPDX-License-Identifier: MIT
 */

//%JXM_CFMT_CFG line-split-by-operator-priority=on

fun primaryTierIf(a: Boolean, b: Boolean, c: Boolean)
{
    if(someVeryLongConditionNameAAAAAAAAAAAAAA
        && anotherVeryLongConditionBBBBBBBBBBBBBBBBBBBBBB
        && yetAnotherLongConditionCCCCCCCCCCCCCCCCCCCCCC) doSomething()
}

fun elvisNeverTreatedAsTernary(x: String?): String
{
    val totalResultValueHereXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX = someVeryLongNullableExpressionNameAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA ?: anotherVeryLongDefaultExpressionNameBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB

    return totalResultValueHereXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
}
